fastlane_core/lib/fastlane_core/PTY.rb

```diff
@@ -0,0 +1,23 @@
+# rubocop:disable Style/FileName
+# via https://github.com/fastlane/fastlane/pull/7202/files
+module FastlaneCore
+  class PTY
+    def self.spawn(*cmd, &block)
+      # We have PTY
+      if Gem::Specification.any? { |s| s.name == "pty" }
+        require "pty"
+        PTY.spawn(command) do |stdout, stdin, pid|
+          block.call(stdin, stdout, pid)
+        end
+      else
+        # We don't - lets try to handle it
+        Actions.verify_gem!('systemu')
```

Comment:
An alternative for those who don't have systemu is to use a fallback like this implementation:

https://github.com/DragonBox/u3d/blob/master/lib/u3d_core/command_runner.rb#L37

Reply:
Uh wow, another way to do this. I'm good with the current situation for now as it is only used in 3 places, lets see if any issues pop up and we can see if this works better then.

May I ask how you found this?

Reply:
It's my code with original implementation of the PTY from fastlane. I use it in u3d which is our project. It's a patch I had meant to submit around 2 years ago to fastlane but I think I forgot...

Reply:
💯  for your implementation!